pkg/backends/calico/client.go

```diff
@@ -173,15 +173,9 @@ func NewCalicoClient(confdConfig *config.Config) (*client, error) {
 	}
 
 	// Create and start route generator.
-	rg, err := NewRouteGenerator(c)
-	if err != nil {
-		log.WithError(err).Fatal("Failed to create route generator")
-	}
-	err = rg.Start()
-	if err != nil {
-		log.WithError(err).Fatal("Failed to start route generator")
+	if rg := NewRouteGenerator(c); rg != nil {
```

Comment:
I think we should do something different here.

If the users _asks_ for static routes but we fail to configure the RG properly, we should fail. 

For example:

```go
if serviceAdvertisementEnabled {
  if err, rg := NewRouteGenerator(c); err != nil {
    log.Error("Error")
    exit(1)
  } else { rg.Start()}
}
```

If it's not enabled, we don't even try to start the route generator.